Sources/Algorithms/Product.swift

```diff
@@ -44,7 +44,8 @@ extension Product2: Sequence {
     }
     
     @inlinable
-    public mutating func next() -> (Base1.Element, Base2.Element)? {
+    public mutating func next() -> (element1: Base1.Element,
+                                    element2: Base2.Element)? {
```

Comment:
Let's keep this tuple unlabeled — `Zip2Sequence`'s element tuple doesn't have labels either, and generic labels such as `element1` and `element2` don't really add any clarity over `.0` and `.1`.

Reply:
I am happy to do this, but I still think that the labels improve readability for users who don't know about tuple indexing. I saw we use more descriptive names for public APIs and type names, so I think it could help users understand the API. Let me know your thoughts.

Reply:
These labels don't improve the users' understanding, because they don't convey any useful information. It's impossible for us to give the tuple elements descriptive names because at this level of abstraction we have nothing to go on. To quote the [API Design Guidelines](https://swift.org/documentation/api-design-guidelines/#naming):

>**Omit needless words.** Every word in a name should convey salient information at the use site.

Feel free to start a discussion about this topic on the forums, as this is equally relevant to the `Zip2Sequence` element type. For the time being we'd like to stick with the precedent set by the standard library.